### PR TITLE
chore: initial setup for staging and prod for iam-did-auth

### PR DESCRIPTION
**Pull request by Timtech4u**

### CODE REVIEW

File: devops/iam-did-auth-proxy-staging/values.yaml [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+            proxy_pass http://did-auth-proxy-helm.iam-did-auth-proxy.svc.cluster.local/healthcheck;
+        }
+
+        location /api {
+            proxy_pass http://gp4btc-backend-microservice.gp4btc.svc.cluster.local;
+        }
+
+        location / {
+            auth_request /token_introspection;
+            proxy_pass http://gp4btc-backend-microservice.gp4btc.svc.cluster.local;
```
Comment (artursudnik) [inline devops/iam-did-auth-proxy-staging/values.yaml:184]: Just doublechecking: is it expected that these hostnames are the same for all environments?

**Review by artursudnik — APPROVED** [review]

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    enabled: false
+    secretRefName: {}
+    configRefName: {}
+
+  appValues:
+    PORT: 80
+    CORS_MAX_AGE: 60
+    CORS_ORIGIN: https://gp4btc-miner.energyweb.org,https://gp4btc-explorer.energyweb.org,http://localhost:4200
+    CACHE_SERVER_LOGIN_PRVKEY: eab5e5ccb983fad7bf7f5cb6b475a7aea95eff0c6523291b0c0ae38b5855459c
+    JWT_SECRET: asecretstring
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:58]: @artursudnik Does this string actual need to be secret? If so, I don't think we can have it in plaintext in a public repo

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+  config:
+    enabled: false
+    secretRefName: {}
+    configRefName: {}
+
+  appValues:
+    PORT: 80
+    CORS_MAX_AGE: 60
+    CORS_ORIGIN: https://gp4btc-miner.energyweb.org,https://gp4btc-explorer.energyweb.org,http://localhost:4200
+    CACHE_SERVER_LOGIN_PRVKEY: eab5e5ccb983fad7bf7f5cb6b475a7aea95eff0c6523291b0c0ae38b5855459c
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:57]: This should be rotated and then put into a sealed secret as well

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    JWT_SECRET: asecretstring
+    JWT_ACCESS_TTL: 3600
+    JWT_REFRESH_TTL: 86400
+    DID_REGISTRY_ADDRESS: 0xc15d5a57a8eb0e1dcbe5d88b8f9a82017e5cc4af
+    ENS_REGISTRY_ADDRESS: 0xd7CeF70Ba7efc2035256d828d5287e2D285CD1ac
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:62]:
```suggestion
    ENS_REGISTRY_ADDRESS: 0x0A6d64413c07E10E890220BBE1c49170080C6Ca0
```

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    ENS_RESOLVER_ADDRESS: 0xcf72f16Ab886776232bea2fcf3689761a0b74EfE
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:63]:
```suggestion
    ENS_RESOLVER_ADDRESS: 0x70ad37DfeB1C05290F4bBd22188FA19Bc154A0ea
```

File: devops/iam-did-auth-proxy-staging/values.yaml [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+    enabled: false
+    secretRefName: {}
+    configRefName: {}
+
+  appValues:
+    PORT: 80
+    CORS_MAX_AGE: 60
+    CORS_ORIGIN: https://gp4btc-miner-stg.energyweb.org,https://gp4btc-explorer-stg.energyweb.org,http://localhost:4200
+    CACHE_SERVER_LOGIN_PRVKEY: eab5e5ccb983fad7bf7f5cb6b475a7aea95eff0c6523291b0c0ae38b5855459c
+    JWT_SECRET: asecretstring
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-staging/values.yaml:58]: In the spirit of making staging as close to prod as possible, these should be rotate and made into sealed secrets

**Review by jrhender — CHANGES_REQUESTED** [review]
Some of the values need to change for prod and hidden for both staging and prod

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+    INCLUDE_ALL_ROLES: false
+    SWAGGER_PATH: api-auth
+    BLOCKNUM_AUTH_ENABLED: true
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:66]: We don't want `BLOCKNUM_AUTH` in `prod`
```suggestion
    BLOCKNUM_AUTH_ENABLED: false
```

File: devops/iam-did-auth-proxy-staging/values.yaml [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+    JWT_ACCESS_TTL: 3600
+    JWT_REFRESH_TTL: 86400
+    DID_REGISTRY_ADDRESS: 0xc15d5a57a8eb0e1dcbe5d88b8f9a82017e5cc4af
+    ENS_REGISTRY_ADDRESS: 0xd7CeF70Ba7efc2035256d828d5287e2D285CD1ac
+    ENS_RESOLVER_ADDRESS: 0xcf72f16Ab886776232bea2fcf3689761a0b74EfE
+    INCLUDE_ALL_ROLES: false
+    SWAGGER_PATH: api-auth
+    BLOCKNUM_AUTH_ENABLED: true
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-staging/values.yaml:66]:
```suggestion
    BLOCKNUM_AUTH_ENABLED: false
```

**Review by jrhender — COMMENTED** [review]
`BLOCKNUM_AUTH_ENABLED` shouldn't be allowed

File: devops/iam-did-auth-proxy-prod/values.yaml (hunk 2 above)
Comment (artursudnik) [reply devops/iam-did-auth-proxy-prod/values.yaml:58]: Yes! Thanks for noticing this! I overlooked it. Anyone can generate an access token and access the backend if it is known to the public.

File: devops/iam-did-auth-proxy-prod/values.yaml (hunk 3 above)
Comment (jrhender) [reply devops/iam-did-auth-proxy-prod/values.yaml:57]: By this, I meant the `CACHE_SERVER_LOGIN_PRVKEY`

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+  appValues:
+    PORT: 80
+    CORS_MAX_AGE: 60
+    CORS_ORIGIN: https://gp4btc-miner.energyweb.org,https://gp4btc-explorer.energyweb.org,http://localhost:4200
+    CACHE_SERVER_LOGIN_PRVKEY: eab5e5ccb983fad7bf7f5cb6b475a7aea95eff0c6523291b0c0ae38b5855459c
+    JWT_SECRET: asecretstring
+    JWT_ACCESS_TTL: 3600
+    JWT_REFRESH_TTL: 86400
+    DID_REGISTRY_ADDRESS: 0xc15d5a57a8eb0e1dcbe5d88b8f9a82017e5cc4af
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:61]:
```suggestion
    DID_REGISTRY_ADDRESS: 0xE29672f34e92b56C9169f9D485fFc8b9A136BCE4
```

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+
+  config:
+    enabled: false
+    secretRefName: {}
+    configRefName: {}
+
+  appValues:
+    PORT: 80
+    CORS_MAX_AGE: 60
+    CORS_ORIGIN: https://gp4btc-miner.energyweb.org,https://gp4btc-explorer.energyweb.org,http://localhost:4200
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:56]: I think it's better to not allow localhost to connect to prod
```suggestion
    CORS_ORIGIN: https://gp4btc-miner.energyweb.org,https://gp4btc-explorer.energyweb.org
```

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+
+  config:
+    enabled: false
+    secretRefName: {}
+    configRefName: {}
+
+  appValues:
+    PORT: 80
+    CORS_MAX_AGE: 60
+    CORS_ORIGIN: https://gp4btc-miner.energyweb.org,https://gp4btc-explorer.energyweb.org
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:56]: I think we should preemptively move to the new domains
```suggestion
    CORS_ORIGIN: https://gp4btc-miner.energyweb.org,https://explorer.gp4btc.org,https://miner.gp4btc.org
```

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+    INCLUDE_ALL_ROLES: false
+    SWAGGER_PATH: api-auth
+    BLOCKNUM_AUTH_ENABLED: false
+    SELF_BASE_URL: https://gp4btc-backend-proxy.energyweb.org
+    SIWE_NONCE_TTL: 300
+    AUTH_COOKIE_ENABLED: true
+    AUTH_COOKIE_SAMESITE_POLICY: none
+
+  opsValues:
+    RPC_URL: https://volta-rpc.energyweb.org/
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:71]:
```suggestion
    RPC_URL: https://rpc.energyweb.org/
```

File: devops/iam-did-auth-proxy-prod/values.yaml [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+    CACHE_SERVER_URL: https://identitycache.energyweb.org/v1
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-prod/values.yaml:72]:
```suggestion
    CACHE_SERVER_URL: https://identitycache-gp4btc.energyweb.org/v1
```

File: devops/iam-did-auth-proxy-staging/values.yaml [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+    SWAGGER_PATH: api-auth
+    BLOCKNUM_AUTH_ENABLED: false
+    SELF_BASE_URL: https://gp4btc-backend-proxy-stg.energyweb.org
+    SIWE_NONCE_TTL: 300
+    AUTH_COOKIE_ENABLED: true
+    AUTH_COOKIE_SAMESITE_POLICY: none
+
+  opsValues:
+    RPC_URL: https://volta-rpc.energyweb.org/
+    CACHE_SERVER_URL: https://identitycache-stg.energyweb.org/v1
```
Comment (jrhender) [inline devops/iam-did-auth-proxy-staging/values.yaml:72]:
```suggestion
    CACHE_SERVER_URL: https://identitycache-gp4btc-staging.energyweb.org/v1
```

Comment (jrhender) [reply devops/iam-did-auth-proxy-staging/values.yaml:72]: @Timtech4u could internal k8s namespace be used here?

File: devops/iam-did-auth-proxy-prod/values.yaml (hunk 13 above)
Comment (jrhender) [reply devops/iam-did-auth-proxy-prod/values.yaml:72]: @Timtech4u could internal k8s namespace be used here?

**Review by jrhender — APPROVED** [review]
Looks good, thanks @Timtech4u

**Pull request merged**